internal/security/secretstoreclient/vault_test.go

```diff
@@ -435,6 +435,204 @@ func TestRevokeSelf(t *testing.T) {
 	code, err := vc.RevokeSelf("fake-token")
 
 	// Assert
-	assert.Nil(err)
+	assert.NoError(err)
+	assert.Equal(http.StatusNoContent, code)
+}
+
+func TestCheckSecretEngineInstalled(t *testing.T) {
+	// Arrange
+	assert := assert.New(t)
+	mockLogger := logger.MockLogger{}
+
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal("GET", r.Method)
+		assert.Equal(VaultMountsAPI, r.URL.EscapedPath())
+		assert.Equal("fake-token", r.Header.Get("X-Vault-Token"))
+
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(`{
+			"data": {
+				"cubbyhole/": {
+					"accessor": "cubbyhole_23676773",
+					"config": {
+						"default_lease_ttl": 0,
+						"force_no_cache": false,
+						"max_lease_ttl": 0
+					},
+					"description": "per-token private secret storage",
+					"local": true,
+					"options": null,
+					"seal_wrap": false,
+					"type": "cubbyhole"
+				},
+				"identity/": {
+					"accessor": "identity_11e23ad0",
+					"config": {
+						"default_lease_ttl": 0,
+						"force_no_cache": false,
+						"max_lease_ttl": 0
+					},
+					"description": "identity store",
+					"local": false,
+					"options": null,
+					"seal_wrap": false,
+					"type": "identity"
+				},
+				"secret/": {
+					"accessor": "kv_3ee7b0c0",
+					"config": {
+						"default_lease_ttl": 0,
+						"force_no_cache": false,
+						"max_lease_ttl": 0
+					},
+					"description": "key/value secret storage",
+					"local": false,
+					"options": {
+						"version": "1"
+					},
+					"seal_wrap": false,
+					"type": "kv"
+				},
+				"sys/": {
+					"accessor": "system_5e0c411d",
+					"config": {
+						"default_lease_ttl": 0,
+						"force_no_cache": false,
+						"max_lease_ttl": 0
+					},
+					"description": "system endpoints used for control, policy and debugging",
+					"local": false,
+					"options": null,
+					"seal_wrap": false,
+					"type": "system"
+				}
+			}	
+		  }`))
+
+	}))
+	defer ts.Close()
+
+	host := strings.Replace(ts.URL, "https://", "", -1)
```

Comment:
if we specify http what should happen?

Reply:
The SecretStoreClient would build a URL of the form "https://https://localhost:#randomport#".   Because we call https.NewTLSServer in the preceeding lines we know that the test HTTP server is actually expecting a TLS connection.